bindings/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# List of language bindings that are assumed to be in corresponding
+# subdirectories of this folder.
+set(binding_LIST "boost_python")
+set(binding_IMPL_TEMPLATE "${CMAKE_CURRENT_SOURCE_DIR}/binding-impl.h.tmpl")
+
+# Iterate over each available language binding and generate a C++ source file.
+foreach(binding_NAME IN LISTS binding_LIST)
+    set(binding_IMPL_OUTPUT "${binding_NAME}_binding-impl.h")
+    set(binding_IMPL_PATH "${CMAKE_CURRENT_SOURCE_DIR}/${binding_NAME}")
+
+    # Generate a binding from file template and add to implementation list,
+    # then append the implementation to variables that are used to create a
+    # single binding library.
+    add_custom_command(OUTPUT "${binding_IMPL_OUTPUT}"
+      COMMAND ${CMAKE_COMMAND}
```

Comment:
I don't know if we actually want to do this or not, I'm worried someone might pass a `${CMAKE_COMMAND}` which is an executable with flags and should _not_ be treated as one argument.  E.g. `/usr/bin/cmake --my-fun-flag`